osx/Updater/Prompt.m

```diff
@@ -0,0 +1,118 @@
+//
+//  Prompt.m
+//  Updater
+//
+//  Created by Gabriel on 4/13/16.
+//  Copyright © 2016 Gabriel Handford. All rights reserved.
+//
+
+#import "Prompt.h"
+#import "TextView.h"
+#import "Defines.h"
+
+@interface FView : NSView
+@end
+
+@implementation Prompt
+
++ (void)showPromptWithInputString:(NSString *)inputString presenter:(NSModalResponse (^)(NSAlert *alert))presenter completion:(void (^)(NSError *error, NSData *output))completion {
+  NSData *data = [inputString dataUsingEncoding:NSUTF8StringEncoding];
+  if (!data) {
+    completion(KBMakeError(@"No data for input"), nil);
+    return;
+  }
+
+  NSError *error = nil;
+  id input = [NSJSONSerialization JSONObjectWithData:data options:0 error:&error];
+  if (!!error) {
+    completion(error, nil);
+    return;
+  }
+  if (!input) {
+    completion(KBMakeError(@"No input for JSON"), nil);
+    return;
+  }
+
+  if (![input isKindOfClass:[NSDictionary class]]) {
+    completion(KBMakeError(@"Invalid input"), nil);
+    return;
+  }
+
+  [self showUpdatePrompt:input presenter:presenter completion:completion];
+}
+
++ (void)showUpdatePrompt:(NSDictionary *)input presenter:(NSModalResponse (^)(NSAlert *alert))presenter completion:(void (^)(NSError *error, NSData *output))completion {
+  NSString *title = [input objectForKey:@"title"];
```

Comment:
Did some refactoring. Added tests for invalid JSON types as values and as root.

I also refactored to have it continue, even with failed inputs instead of exiting with a non-zero status.